src/helper-modules/govuk-notify/govuk-notify.service.test.ts

```diff
@@ -0,0 +1,140 @@
+      const response = await service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(response).toEqual({ status: 201, data: expectedResponse });
+    });
+
+    it('calls GOV.UK Notify client with the specified field `reference` and returns a 201 response', async () => {
+      const reference = valueGenerator.string({ length: 10 });
+      const response = await service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation, reference });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledWith(templateId, sendToEmailAddress, { personalisation, reference });
+      expect(response).toEqual({ status: 201, data: expectedResponse });
+    });
+
+    it.each([
+      {
+        exceptionClass: BadRequestException,
+        exceptionName: 'Bad Request Exception',
+        error: 'Bad Request',
+        status: 400,
+      },
+      {
+        exceptionClass: UnauthorizedException,
+        exceptionName: 'Unauthorized Exception',
+        error: 'Unauthorized',
+        status: 401,
+      },
+      {
+        exceptionClass: ForbiddenException,
+        exceptionName: 'Forbidden Exception',
+        error: 'Forbidden',
+        status: 403,
+      },
+      {
+        exceptionClass: InternalServerErrorException,
+        exceptionName: 'Internal Server Error Exception',
+        error: 'Internal Server Error',
+        status: 500,
+      },
+    ])('throws exception $exceptionName for unexpected $status', async ({ exceptionClass, exceptionName, error, status }) => {
+      jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(status, errorMessage)));
+
+      const resultPromise = service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      await expect(resultPromise).rejects.toBeInstanceOf(exceptionClass);
+      await expect(resultPromise).rejects.toThrow(exceptionName);
+      await expect(resultPromise).rejects.toHaveProperty('status', status);
+      await expect(resultPromise).rejects.toHaveProperty('response', { message: [errorMessage], error, statusCode: status });
+    });
+
+    it('throws generic Error exception for unexpected status', async () => {
+      const unexpectedStatus = valueGenerator.integer({ min: 900, max: 999 });
+      jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(unexpectedStatus, errorMessage)));
+
+      const resultPromise = service.sendEmail(govUkNotifyKey, { sendToEmailAddress, templateId, personalisation });
+
+      expect(sendEmailMethodMock).toHaveBeenCalledTimes(1);
+      await expect(resultPromise).rejects.toBeInstanceOf(Error);
+      await expect(resultPromise).rejects.toThrow(errorMessage);
+      await expect(resultPromise).rejects.toHaveProperty('message', errorMessage);
+      await expect(resultPromise).rejects.toHaveProperty('stack');
```

Comment:
Why do we check that the `resultPromise` has the `'stack'` property? It looks like this property isn't defined in the JS spec yet https://stackoverflow.com/questions/75109188/for-what-reason-stack-property-is-optional-in-error. Clearly it must be implemented in Node, but I don't think there's any value in checking for this.

Reply:
It is documented in NodeJS, but we usually don't check it. Removed.